src/main/java/com/commercetools/sync/services/impl/ProductTypeServiceImpl.java

```diff
@@ -61,8 +63,10 @@ public ProductTypeServiceImpl(@Nonnull final BaseSyncOptions syncOptions) {
                     if (StringUtils.isNotBlank(fetchedTypeKey)) {
                         keyToIdCache.put(fetchedTypeKey, id);
                     } else {
-                        syncOptions.applyWarningCallback(format("ProductType with id: '%s' has no key set. Keys are"
-                                + " required for productType matching.", id));
+                        // syncOptions is ProductSyncOptions
+                        syncOptions.applyWarningCallback(
+                            new SyncException(format("ProductType with id: '%s' has no key set. Keys are"
+                                + " required for productType matching.", id)), null, null);
```

Comment:
Because we don't have it here. It fetches list of `ProductType` and it is using `ProductSyncOptions`. Types don't match (`ProductType` is not `Product`).

Reply:
I guess in this case we can pass `productType` to the callback as `oldResource` param.